File: Data Structures & Algorithms/max-water-container/submission-2.java
class Solution {
    public int maxArea(int[] heights) {
        int result = 0;
        int left = 0;
        int right = heights.length - 1;

        while (left < right) {
            int area = calculateArea(left, right, heights);
            result = Math.max(result, area);

            if (heights[left] < heights[right]) {
                left++;
            } else {
                right--;
            }
        }
        return result;
    }


    private int calculateArea(int left, int right, int[] heights) {
        int height = Math.min(heights[left], heights[right]);
        return (right - left) * height;
        
    }
}

File: Data Structures & Algorithms/merge-intervals/submission-2.ts
class Solution {
    /**
     * @param {number[][]} intervals
     * @return {number[][]}
     */
    merge(intervals: number[][]): number[][] {
        intervals.sort((a,b) => a[0] - b[0]);
        const result: number[][] = [];
        result.push(intervals[0]);

        for (let i = 1; i < intervals.length; i++) {
            if (intervals[i][0] <= result[result.length - 1][1]) {
                result[result.length - 1][0] = Math.min(result[result.length - 1][0], intervals[i][0]);
                result[result.length - 1][1] = Math.max(result[result.length - 1][1], intervals[i][1]);
            } else {
                result.push(intervals[i]);
            }
        }

        return result;
    }
}
